mepo.d/state/state.py

```diff
@@ -56,6 +59,13 @@ def initialize(cls, project_config_file):
         input_components = ConfigFile(project_config_file).read_file()
         complist = list()
         for name, comp in input_components.items():
+            for key, value in comp.items():
+                if key == "remote":
+                    if comp[key].startswith('..'):
+                        rel_remote = os.path.basename(comp[key])
+                        fixture_url = get_current_remote_url()
+                        resolved_remote = urljoin(fixture_url,rel_remote)
+                        comp[key] = resolved_remote
```

Comment:
I see what you mean about nested loops.   It does seem there ought to be a more direct way to do this in Python.   But I have no idea how.